server-spi/src/main/java/org/keycloak/storage/group/GroupLookupProvider.java

```diff
@@ -81,4 +79,15 @@ default Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String s
      */
     Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String search, Boolean exact, Integer firstResult, Integer maxResults);
 
+    /**
+     * Returns the subgroups of a given group id
+     * @param realm Realm
+     * @param id the id of parent group to be used to load subgroups with
+     * @param firstResult First result to return, Ignored if negative or {@code null}.
+     * @param maxResults Maximum number of results to return. Ignored if negative or {@code null}.
+     * @return A Stream of groups that all have the given parent group as their parent.
+     */
+    Stream<GroupModel> searchForSubgroupsByParentIdStream(RealmModel realm, String id, Integer firstResult, Integer maxResults);
```

Comment:
This one can be a `default that just delegates to the other method but passes `null` to the `search|name` parameter. It should also remove unnecessary overrides down the hierarchy.

Reply:
addressed this in the current commit